### PR TITLE
Allow error customization for all errors

### PR DESCRIPTION
**Pull request by yufeih**
Currently we only allow custom error code, severity and message for schema validation errors, this PR extend the support to all errors.

#4927 

Docs.Build changes: https://dev.azure.com/ceapex/Engineering/_git/Docs.Build/pullrequest/14134?_a=overview

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4933)

### CODE REVIEW

File: src/docfx/lib/json/ShortHandConverter.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            switch (reader.TokenType)
+            {
+                case JsonToken.Null:
+                case JsonToken.Undefined:
+                    return null;
+
+                case JsonToken.StartObject:
+                case JsonToken.StartArray:
```
Comment (mingwli) [inline src/docfx/lib/json/ShortHandConverter.cs]: > JsonToken.StartArray [](start = 20, length = 21)

when would a user enter such case 🙋🏽‍♂️

File: test/docfx.Test/lib/JsonSchemaTest.cs [hunk 2]
```diff
@@ -330,13 +330,13 @@ public void TestJsonSchemaConfirmance(string description, string schemaText, str
             "['warning','invalid-value','Invalid value for 'key1[0]': 'yyy'','file',1,15]")]
 
         // additional errors
-        [InlineData("{'required': ['author'], 'additionalErrors': {'author': {'missing-attribute': {'severity': 'suggestion', 'code': 'author-missing', 'message': 'Add a valid GitHub ID.'}}}}", "{'b': 1}",
+        [InlineData("{'required': ['author'], 'customErrors': {'author': {'missing-attribute': {'severity': 'suggestion', 'code': 'author-missing', 'additionalMessage': 'Add a valid GitHub ID.'}}}}", "{'b': 1}",
```
Comment (LowEntropyBody) [inline test/docfx.Test/lib/JsonSchemaTest.cs]: Change docs build too, from 'additionalErrors' to 'customErrors'

Comment (yufeih) [reply test/docfx.Test/lib/JsonSchemaTest.cs]: https://dev.azure.com/ceapex/Engineering/_git/Docs.Build/pullrequest/14134?_a=overview

File: src/docfx/lib/json/ShortHandConverter.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace Microsoft.Docs.Build
+{
+    internal class ShortHandConverter : JsonConverter
```
Comment (LowEntropyBody) [inline src/docfx/lib/json/ShortHandConverter.cs]: Add comments for explanation

**Pull request merged**